chirpstack/src/maccommand/mod.rs

```diff
@@ -126,6 +133,156 @@ pub async fn handle_uplink<'a>(
     Ok((out, must_respond_with_downlink))
 }
 
+// Checks for MAC commands which the device is unresponsive towards.
+pub async fn check_nonresponsive_mac_commands(
+    uplink_frame_set: &UplinkFrameSet,
+    app: &application::Application,
+    dev: &device::Device,
+    di: &integration_pb::DeviceInfo,
+    ds: &mut internal::DeviceSession,
+    is_relay: bool,
+) {
+    // A list of MAC commands that the device is expected to respond to, if pending.
+    let mut cids: Vec<lrwn::CID> = Vec::from([
```

Comment:
My worry is that it will be very likely that this list will get out-of-sync when new mac-commands are introduced.

Reply:
But not all MAC commands require a response, so I realize that you need to have a fixed list somewhere, in some form. Presently, Chirpstack has fixed code for handling responses to MAC commands, so I could not see how we can completely avoid hardcoding this list.

Perhaps what I can do differently here, is to add another function call for the MAC commands, allowing each MAC command implementation to specify whether a response is expected or not. For example, NewChannelReq could indicate that it expects a response, for the mod.rs to take action if there is no response. I'll try to come up with a new implementation and show it to you.

Thank you for taking the time to review my work.